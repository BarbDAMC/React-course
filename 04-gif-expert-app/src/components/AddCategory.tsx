import { useState } from "react"

interface AddCategoryProps {
  onNewCategory: (inputValue: string) => void;
}

export const AddCategory = ({onNewCategory} : AddCategoryProps) => {
  const [inputValue, setInputValue] = useState('One Punch');
  
  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => setInputValue(e.target.value);
  
  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    const newInputValue = inputValue.trim()
    if(newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input 
        type="text"
        name="gifs-input"
        value={inputValue}
        placeholder="Buscar gifs"
        onChange={onInputChange}
      />
    </form>
  )
}
