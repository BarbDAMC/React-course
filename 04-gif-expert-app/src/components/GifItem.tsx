import type { Image } from "../helpers/gifImageInterface"


export const GifItem = ({ title, url }: Image) => {

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}
