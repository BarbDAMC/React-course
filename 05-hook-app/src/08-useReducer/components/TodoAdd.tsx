import { useState } from "react";
import { ACTION_TYPES, type todoInterface } from "../interfaces";

interface todoAddInterface {
  onNewTodo: (todo: todoInterface, actionType: string) => void;
}

export const TodoAdd = ({onNewTodo}: todoAddInterface) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    const newTodo = {
      id: new Date().getTime() * 4,
      description: inputValue,
      done: false
    }

    onNewTodo(newTodo, ACTION_TYPES.ADD_TODO);
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text" 
        placeholder="¿Qué hay que hacer?"
        className="form-control"
        value={inputValue}
        onChange={handleChange}
      />

      <button
        type="submit"
        className="btn btn-outline-primary mt-1"
      >
        Agregar
      </button>
    </form>
  )
}
