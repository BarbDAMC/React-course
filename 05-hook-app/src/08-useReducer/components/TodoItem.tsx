import { ACTION_TYPES, type todoInterface } from "../interfaces"

interface todoItemInterface {
  todo: todoInterface,
  onActionTodo: (todo: todoInterface, actionType: string) => void,
}

export const TodoItem = ({todo, onActionTodo}: todoItemInterface) => {
  return (
    <li className="list-group-item d-flex justify-content-between">
      <span 
        className={`align-self-center ${todo.done ? 'text-decoration-line-through' : ''}`}
        onClick={() => onActionTodo(todo, ACTION_TYPES.TOGGLE_TODO)}
      >
        {todo.description}
      </span>
      <button 
        className="btn btn-danger"
        onClick={() => onActionTodo(todo, ACTION_TYPES.REMOVE_TODO)}
      >
        Borrar
      </button>
    </li>
  )
}
