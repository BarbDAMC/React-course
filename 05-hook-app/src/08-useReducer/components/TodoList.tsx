import type { todoInterface } from "../interfaces"
import { TodoItem } from "./TodoItem"

interface todoListInterface {
  todos: todoInterface[],
  onActionTodo: (todo: todoInterface, actionType: string) => void,
}

export const TodoList = ({todos, onActionTodo}: todoListInterface) => {
  return (
    <ul className="list-group">
      {
        todos.map((todo: todoInterface) => (
          <TodoItem key={todo.id} todo={todo} onActionTodo={onActionTodo} />
        ))
      }
            
    </ul>
  )
}
