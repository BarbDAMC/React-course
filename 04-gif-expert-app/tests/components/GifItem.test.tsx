import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('<GifItem />', () => {
  const testingImage = {
    id: 'testing-id',
    url: 'https://testing-url.png',
    title: 'testing-title'
  }


  test('Shoul match with the snashot', () => {
    const { container } = render(<GifItem {...testingImage}/>);
    expect(container).toMatchSnapshot();
  })

  test('should show the image with the defined url and alt text', () => {
    render(<GifItem {...testingImage}/>);

    const {src, alt} = screen.getByRole('img') as HTMLInputElement;

    expect(src).toBe('https://testing-url.png/');
    expect(alt).toBe('testing-title');
  })

  test('should render the title as a paragraph', () => {
    render(<GifItem {...testingImage}/>);

    expect(screen.getByText('testing-title')).toBeTruthy();
  })
  
  

})
