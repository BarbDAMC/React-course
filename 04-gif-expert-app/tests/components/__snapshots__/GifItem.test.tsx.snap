// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<GifItem /> Shoul match with the snashot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="testing-title"
      src="https://testing-url.png"
    />
    <p>
      testing-title
    </p>
  </div>
</div>
`;
